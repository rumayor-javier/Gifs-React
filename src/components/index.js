export * from "./GifCard";
export * from "./GifGrid";
export * from "./SearchInput";