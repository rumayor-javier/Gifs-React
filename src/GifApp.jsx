import { useState } from "react";
import { GifGrid, SearchInput } from "./components";

export const GifApp = () => {
  const [queries, setQuery] = useState(["Goku"]);

  const onAddQuery = (newQuery) => {
    if (queries.includes(newQuery)) return;
    setQuery((queries) => [newQuery, ...queries]);
  };

  return (
    <>
      <h1>GifApp</h1>

      <SearchInput onAddQuery={onAddQuery} />

      {queries.map((query) => (
        <GifGrid key={query} query={query} />
      ))}
    </>
  );
};
