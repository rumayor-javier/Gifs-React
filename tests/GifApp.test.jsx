import { render, screen } from "@testing-library/react";
import { GifApp } from "../src/GifApp";

describe("Pruebas a <GifApp />", () => {
  test("Debe contener un titulo", () => {
    const { container } = render(<GifApp />);

    expect(container).toMatchSnapshot();
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe("GifApp");
  });
});
