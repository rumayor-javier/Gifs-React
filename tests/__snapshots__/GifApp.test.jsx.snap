// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifApp /> Debe contener un titulo 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gifs"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
  <h3>
    Goku
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
